src/app/(auth)/user/page.tsx

```diff
@@ -1,46 +1,62 @@
 'use client';
 
-import { getUsers } from '@/apis/user';
+import { getUsers, deleteUser } from '@/apis/user';
 import Table from '@/components/Table';
-import Button from '@/components/Button';
-import { Users, TableData } from '@/utils/types';
+import { UserResource, TableData } from '@/utils/types';
 import { useEffect, useState } from 'react';
 import { PencilSquareIcon, TrashIcon } from '@heroicons/react/24/solid';
 import { UsersIcon } from '@heroicons/react/24/outline';
-// import Input from '@/components/Input';
+import Link from 'next/link';
+import { toast } from 'react-toastify';
 
 const Page = () => {
-  const [users, setUsers] = useState<Users | undefined>(undefined);
+  const [users, setUsers] = useState<UserResource | undefined>(undefined);
 
   const [tableData, setTableData] = useState<TableData>({
     headers: ['Name', 'Email'],
     rows: [],
   });
 
+  const fetchUsers = async () => {
+    setUsers(await getUsers());
+  };
+
   useEffect(() => {
-    const fetchUsers = async () => {
-      setUsers(await getUsers());
-    };
     fetchUsers();
-  }, []);
+    const showDeleteConfirmationModal = async (id: number) => {
+      const result = confirm('Are you sure you want to delete this user?');
+      if (result) {
+        try {
+          await deleteUser(id);
+          toast.success('User deleted successfully');
+          fetchUsers();
+        } catch (error: any) {
+          toast.error(error.response.data.message);
+        }
+      }
+    };
 
-  useEffect(() => {
     if (users) {
-      const rows = users.data.map((user: Users) => {
+      const rows = users.data.map((user: any) => {
         return [
           <>
             <div>{user.name}</div>
-            {/* <div className="text-gray-500">{campaign.tag.name}</div> */}
-          </>,
-          <>
             <div>{user.email}</div>
-          </>,
-          <>
             <div className="flex justify-end">
-              <a href="#" className="text-gray-400 hover:text-indigo-700">
+              <Link
+                href={{
+                  pathname: '/user/edit/',
+                  query: { id: user.id },
```

Comment:
just to confirm, the edit page url should be something like this:
```
/user/12/edit
```
where 12 is the id of the user.

In case it's not like that, I would recommend using route params and modifying the route.